### ISSUE TITLE
Harbor 2 support

### ISSUE THREAD

**Issue by Vad1mo** [c4po/harbor_exporter#15]
Would be good if Harbor 2.x was supported or is it?

**Comment by c4po** [comment]
yes, I'm checking the changes in 2.x to see any impact on the metric api.
also I was hoping the exporter can be included in Harbor natively

**Comment by Vad1mo** [comment]
Harbor exposing metrics by itself would be nice. On the other hand an exporter in a good helm chart es easy to install and use.

**Comment by tetra-geoffrey-byers** [comment]
I have created a pull request (#19) to update the API endpoints to the 2.0 versions. Seems to work okay so far.

**Comment by badsmoke** [comment]
cannot yet confirm that it runs completely with v2:

my one project has 26 repos
but only 10 of these are displayed in the exporter.

"total_repos" is displayed correctly

tags do not match:
gui: 19 tags/artifacts   
exporter: 0tags/artifacts


built the exporter with the help of the pull request (#19)

**Comment by c4po** [comment]
I just add the feature to detect Harbor api version. So it should automatically support Harbor v1 and v2 apis.
new version is v0.4.0. Please test and let me know if it has any issues

**Comment by badsmoke** [comment]
does not work for me neither with version harbor 2.0.0 nor 2.0.1

only the repo metrics are displayed



```
# HELP go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 1.808e-05
go_gc_duration_seconds{quantile="0.25"} 3.3811e-05
go_gc_duration_seconds{quantile="0.5"} 0.000132043
go_gc_duration_seconds{quantile="0.75"} 0.000841879
go_gc_duration_seconds{quantile="1"} 0.010100488
go_gc_duration_seconds_sum 0.019579662
go_gc_duration_seconds_count 21
# HELP go_goroutines Number of goroutines that currently exist.
# TYPE go_goroutines gauge
go_goroutines 15
# HELP go_info Information about the Go environment.
# TYPE go_info gauge
go_info{version="go1.13.12"} 1
# HELP go_memstats_alloc_bytes Number of bytes allocated and still in use.
# TYPE go_memstats_alloc_bytes gauge
go_memstats_alloc_bytes 3.82388e+06
# HELP go_memstats_alloc_bytes_total Total number of bytes allocated, even if freed.
# TYPE go_memstats_alloc_bytes_total counter
go_memstats_alloc_bytes_total 5.2747704e+07
# HELP go_memstats_buck_hash_sys_bytes Number of bytes used by the profiling bucket hash table.
# TYPE go_memstats_buck_hash_sys_bytes gauge
go_memstats_buck_hash_sys_bytes 1.453194e+06
# HELP go_memstats_frees_total Total number of frees.
# TYPE go_memstats_frees_total counter
go_memstats_frees_total 302617
# HELP go_memstats_gc_cpu_fraction The fraction of this program's available CPU time used by the GC since the program started.
# TYPE go_memstats_gc_cpu_fraction gauge
go_memstats_gc_cpu_fraction 7.454566071289779e-05
# HELP go_memstats_gc_sys_bytes Number of bytes used for garbage collection system metadata.
# TYPE go_memstats_gc_sys_bytes gauge
go_memstats_gc_sys_bytes 2.38592e+06
# HELP go_memstats_heap_alloc_bytes Number of heap bytes allocated and still in use.
# TYPE go_memstats_heap_alloc_bytes gauge
go_memstats_heap_alloc_bytes 3.82388e+06
# HELP go_memstats_heap_idle_bytes Number of heap bytes waiting to be used.
# TYPE go_memstats_heap_idle_bytes gauge
go_memstats_heap_idle_bytes 6.0448768e+07
# HELP go_memstats_heap_inuse_bytes Number of heap bytes that are in use.
# TYPE go_memstats_heap_inuse_bytes gauge
go_memstats_heap_inuse_bytes 5.873664e+06
# HELP go_memstats_heap_objects Number of allocated objects.
# TYPE go_memstats_heap_objects gauge
go_memstats_heap_objects 13124
# HELP go_memstats_heap_released_bytes Number of heap bytes released to OS.
# TYPE go_memstats_heap_released_bytes gauge
go_memstats_heap_released_bytes 5.9523072e+07
# HELP go_memstats_heap_sys_bytes Number of heap bytes obtained from system.
# TYPE go_memstats_heap_sys_bytes gauge
go_memstats_heap_sys_bytes 6.6322432e+07
# HELP go_memstats_last_gc_time_seconds Number of seconds since 1970 of last garbage collection.
# TYPE go_memstats_last_gc_time_seconds gauge
go_memstats_last_gc_time_seconds 1.5936763164435601e+09
# HELP go_memstats_lookups_total Total number of pointer lookups.
# TYPE go_memstats_lookups_total counter
go_memstats_lookups_total 0
# HELP go_memstats_mallocs_total Total number of mallocs.
# TYPE go_memstats_mallocs_total counter
go_memstats_mallocs_total 315741
# HELP go_memstats_mcache_inuse_bytes Number of bytes in use by mcache structures.
# TYPE go_memstats_mcache_inuse_bytes gauge
go_memstats_mcache_inuse_bytes 10416
# HELP go_memstats_mcache_sys_bytes Number of bytes used for mcache structures obtained from system.
# TYPE go_memstats_mcache_sys_bytes gauge
go_memstats_mcache_sys_bytes 16384
# HELP go_memstats_mspan_inuse_bytes Number of bytes in use by mspan structures.
# TYPE go_memstats_mspan_inuse_bytes gauge
go_memstats_mspan_inuse_bytes 67184
# HELP go_memstats_mspan_sys_bytes Number of bytes used for mspan structures obtained from system.
# TYPE go_memstats_mspan_sys_bytes gauge
go_memstats_mspan_sys_bytes 81920
# HELP go_memstats_next_gc_bytes Number of heap bytes when next garbage collection will take place.
# TYPE go_memstats_next_gc_bytes gauge
go_memstats_next_gc_bytes 5.854112e+06
# HELP go_memstats_other_sys_bytes Number of bytes used for other system allocations.
# TYPE go_memstats_other_sys_bytes gauge
go_memstats_other_sys_bytes 1.502318e+06
# HELP go_memstats_stack_inuse_bytes Number of bytes in use by the stack allocator.
# TYPE go_memstats_stack_inuse_bytes gauge
go_memstats_stack_inuse_bytes 786432
# HELP go_memstats_stack_sys_bytes Number of bytes obtained from system for stack allocator.
# TYPE go_memstats_stack_sys_bytes gauge
go_memstats_stack_sys_bytes 786432
# HELP go_memstats_sys_bytes Number of bytes obtained from system.
# TYPE go_memstats_sys_bytes gauge
go_memstats_sys_bytes 7.25486e+07
# HELP go_threads Number of OS threads created.
# TYPE go_threads gauge
go_threads 12
# HELP harbor_exporter_build_info A metric with a constant '1' value labeled by version, revision, branch, and goversion from which harbor_exporter was built.
# TYPE harbor_exporter_build_info gauge
harbor_exporter_build_info{branch="master",goversion="go1.13.12",revision="e04c7f853dadba636fd49dc1500cb53bea87f2b9",version="v0.4.0"} 1
# HELP harbor_project_count_total projects number relevant to the user
# TYPE harbor_project_count_total gauge
harbor_project_count_total{type="private_project"} 3
harbor_project_count_total{type="public_project"} 2
harbor_project_count_total{type="total_project"} 5
# HELP harbor_quotas_count_total quotas
# TYPE harbor_quotas_count_total gauge
harbor_quotas_count_total{repo_id="1",repo_name="library",type="hard"} 0
harbor_quotas_count_total{repo_id="1",repo_name="library",type="used"} 0
harbor_quotas_count_total{repo_id="2",repo_name="apps",type="hard"} 0
harbor_quotas_count_total{repo_id="2",repo_name="apps",type="used"} 0
harbor_quotas_count_total{repo_id="3",repo_name="dev",type="hard"} 0
harbor_quotas_count_total{repo_id="3",repo_name="dev",type="used"} 0
harbor_quotas_count_total{repo_id="4",repo_name="public",type="hard"} 0
harbor_quotas_count_total{repo_id="4",repo_name="public",type="used"} 0
harbor_quotas_count_total{repo_id="5",repo_name="armv8",type="hard"} 0
harbor_quotas_count_total{repo_id="5",repo_name="armv8",type="used"} 0
# HELP harbor_quotas_size_bytes quotas
# TYPE harbor_quotas_size_bytes gauge
harbor_quotas_size_bytes{repo_id="1",repo_name="library",type="hard"} -1
harbor_quotas_size_bytes{repo_id="1",repo_name="library",type="used"} 0
harbor_quotas_size_bytes{repo_id="2",repo_name="apps",type="hard"} 5.36870912e+11
harbor_quotas_size_bytes{repo_id="2",repo_name="apps",type="used"} 6.34969361e+09
harbor_quotas_size_bytes{repo_id="3",repo_name="dev",type="hard"} 5.36870912e+11
harbor_quotas_size_bytes{repo_id="3",repo_name="dev",type="used"} 2.7738157461e+10
harbor_quotas_size_bytes{repo_id="4",repo_name="public",type="hard"} 1.2884901888e+11
harbor_quotas_size_bytes{repo_id="4",repo_name="public",type="used"} 2.07606739e+08
harbor_quotas_size_bytes{repo_id="5",repo_name="armv8",type="hard"} -1
harbor_quotas_size_bytes{repo_id="5",repo_name="armv8",type="used"} 3.5652777e+07
# HELP harbor_replication_status Get status of the last execution of this replication policy: Succeed = 1, any other status = 0.
# TYPE harbor_replication_status gauge
harbor_replication_status{repl_pol_name="arm-test"} 1
harbor_replication_status{repl_pol_name="docker-hub"} 1
harbor_replication_status{repl_pol_name="docker-apps"} 1
harbor_replication_status{repl_pol_name="docker-dev"} 1
# HELP harbor_replication_tasks Get number of replication tasks, with various results, in the latest execution of this replication policy.
# TYPE harbor_replication_tasks gauge
harbor_replication_tasks{repl_pol_name="arm-test",result="failed"} 0
harbor_replication_tasks{repl_pol_name="arm-test",result="in_progress"} 0
harbor_replication_tasks{repl_pol_name="arm-test",result="stopped"} 0
harbor_replication_tasks{repl_pol_name="arm-test",result="succeed"} 1
harbor_replication_tasks{repl_pol_name="docker-hub",result="failed"} 0
harbor_replication_tasks{repl_pol_name="docker-hub",result="in_progress"} 0
harbor_replication_tasks{repl_pol_name="docker-hub",result="stopped"} 0
harbor_replication_tasks{repl_pol_name="docker-hub",result="succeed"} 0
harbor_replication_tasks{repl_pol_name="docker-apps",result="failed"} 0
harbor_replication_tasks{repl_pol_name="docker-apps",result="in_progress"} 0
harbor_replication_tasks{repl_pol_name="docker-apps",result="stopped"} 0
harbor_replication_tasks{repl_pol_name="docker-apps",result="succeed"} 23
harbor_replication_tasks{repl_pol_name="docker-dev",result="failed"} 0
harbor_replication_tasks{repl_pol_name="docker-dev",result="in_progress"} 0
harbor_replication_tasks{repl_pol_name="docker-dev",result="stopped"} 0
harbor_replication_tasks{repl_pol_name="docker-dev",result="succeed"} 31
# HELP harbor_repo_count_total repositories number relevant to the user
# TYPE harbor_repo_count_total gauge
harbor_repo_count_total{type="private_repo"} 55
harbor_repo_count_total{type="public_repo"} 3
harbor_repo_count_total{type="total_repo"} 58
# HELP harbor_scans_completed metrics of the latest scan all process
# TYPE harbor_scans_completed gauge
harbor_scans_completed 0
# HELP harbor_scans_requester metrics of the latest scan all process
# TYPE harbor_scans_requester gauge
harbor_scans_requester 0
# HELP harbor_scans_total metrics of the latest scan all process
# TYPE harbor_scans_total gauge
harbor_scans_total 0
# HELP harbor_system_volumes_bytes Get system volume info (total/free size).
# TYPE harbor_system_volumes_bytes gauge
harbor_system_volumes_bytes{storage="free"} 5.18154932224e+11
harbor_system_volumes_bytes{storage="total"} 5.49753716736e+11
# HELP harbor_up Was the last query of harbor successful.
# TYPE harbor_up gauge
harbor_up 0
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total counter
process_cpu_seconds_total 2.9
# HELP process_max_fds Maximum number of open file descriptors.
# TYPE process_max_fds gauge
process_max_fds 1.048576e+06
# HELP process_open_fds Number of open file descriptors.
# TYPE process_open_fds gauge
process_open_fds 11
# HELP process_resident_memory_bytes Resident memory size in bytes.
# TYPE process_resident_memory_bytes gauge
process_resident_memory_bytes 1.8649088e+07
# HELP process_start_time_seconds Start time of the process since unix epoch in seconds.
# TYPE process_start_time_seconds gauge
process_start_time_seconds 1.59367592208e+09
# HELP process_virtual_memory_bytes Virtual memory size in bytes.
# TYPE process_virtual_memory_bytes gauge
process_virtual_memory_bytes 1.16682752e+08
# HELP process_virtual_memory_max_bytes Maximum amount of virtual memory available in bytes.
# TYPE process_virtual_memory_max_bytes gauge
process_virtual_memory_max_bytes -1
# HELP promhttp_metric_handler_requests_in_flight Current number of scrapes being served.
# TYPE promhttp_metric_handler_requests_in_flight gauge
promhttp_metric_handler_requests_in_flight 1
# HELP promhttp_metric_handler_requests_total Total number of scrapes by HTTP status code.
# TYPE promhttp_metric_handler_requests_total counter
promhttp_metric_handler_requests_total{code="200"} 38
promhttp_metric_handler_requests_total{code="500"} 0
promhttp_metric_handler_requests_total{code="503"} 0

```

**Comment by c4po** [comment]
the `repositories_tags_total` metric should be fixed with `v0.5.0`
will have another PR to fix the page size issue
